cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -1243,10 +1389,11 @@ impl MachInst for Inst {
         Inst::jmp_known(BranchTarget::Label(label))
     }
 
-    fn gen_constant(to_reg: Writable<Reg>, value: u64, _: Type) -> SmallVec<[Self; 4]> {
+    fn gen_constant(to_reg: Writable<Reg>, value: u64, ty: Type) -> SmallVec<[Self; 4]> {
         let mut ret = SmallVec::new();
-        let is64 = value > 0xffff_ffff;
-        ret.push(Inst::imm_r(is64, value, to_reg));
+        debug_assert!(ty.is_int(), "float constants NYI");
+        let is_64 = ty == I64 && value > 0x7fffffff;
```

Comment:
Perhaps for a future refactor, but IMHO we should move from `is_64` bools to a purpose-built enum.

Reply:
Yes! Recent PRs with `OperandSize` enum have inspired me... Added to the backlog!